### ISSUE TITLE
Heartbeat MQTT Message

### ISSUE THREAD

**Issue by hmmbob** [andrewjfreyer/monitor#20]
Would it be possible to implement a heartbeat message? I would like to use this to check if the service is running on the various devices. 

An idea on implementation would be 
`Topic: $mqtt_topicpath/heartbeat
Payload: "{\"identity\":\"$mqtt_publisher_identity\"}"`

In homeassistant, one could use this single message (and the state.last_updated metadata of the message) to see if updates are still coming in. I find my (remote) Pi Zero W somewhat unreliable in keeping the WiFi alive, so with an automation I can toggle power from the main Pi to give the Zero a reboot.

**Comment by andrewjfreyer** [comment]
Yes, we could add a heartbeat. However, the preferred method for detecting downed nodes is with a last-will message, which is sent whenever a device goes offline. Do you have a use case outside of detecting downed nodes?

**Comment by hmmbob** [comment]
Nope, it is really the “is it still alive” use case. I was working with the -e option to abuse that message as a heartbeat signal, when I ran into the other issue I co-reported.

To be sure: a last will message will also be set if the device goes offline due to (i.e.) wlan0 going down? Should we also set a birth message in the same topic as the LWT? Maybe:

`Payload: "{\"$mqtt_publisher_identity\":\"Online\"}"` and `Payload: "{\"$mqtt_publisher_identity\":\"Offline\"}"`

**Comment by andrewjfreyer** [comment]
Yes, `offline` is the message released by the broker when the connection with a node is broken. So the last will will be sent no matter what - regardless whether the node exits gracefully, whether lan goes down, whatever. 

I'll add an option for heartbeat.

**Comment by andrewjfreyer** [comment]
Also, good catch on the "online" message - I removed that for some reason; need to add it back.

**Comment by hmmbob** [comment]
Thanks!

Off-topic, but I wanted to let you know I am really happy with this evolution of presence; that project was giving me other issues with 2,4 WiFi (my wife’s phone just froze from time to time) and BT (hers didn’t connect to anything anymore, mine lost connection with my smart watch) but this monitor project is working like a charm for me!

**Comment by andrewjfreyer** [comment]
Hey, that's GREAT to hear. I had a few alpha testers of monitor who ended up switching to 5GHz networks while I was developing! Their solution to interference avoidance didn't need monitor at all, haha. 

I'm very pleased to hear that this has cleared your interference. I have set up a test 2.4GHz network, and I was seeing less interference, but it's always hard to tell how similar a test environment will be to an actual use case. 

Really motivating to hear that the time I've put into this complete re-write of something that was working fine for me (5GHz main network, haha) is worth it. Thanks!

As a side note, I'm going to close this issue because I have added both the heartbeat and the online message for mqtt.
